step-02/js/main.js

```diff
@@ -1,239 +1,278 @@
-  pc2.createAnswer().then(
-    onCreateAnswerSuccess,
-    onCreateSessionDescriptionError
-  );
-}
+  // Create peer connections and add behavior.
+  localPeerConnection = new RTCPeerConnection(servers);
+  trace('Created local peer connection object localPeerConnection.');
 
-function onSetLocalSuccess(pc) {
-  trace(getName(pc) + ' setLocalDescription complete');
-}
+  localPeerConnection.addEventListener('icecandidate', connectionHandler);
+  localPeerConnection.addEventListener(
+    'iceconnectionstatechange', connectionChanged);
 
-function onSetRemoteSuccess(pc) {
-  trace(getName(pc) + ' setRemoteDescription complete');
-}
+  remotePeerConnection = new RTCPeerConnection(servers);
+  trace('Created remote peer connection object remotePeerConnection.');
 
-function onSetSessionDescriptionError(error) {
-  trace('Failed to set session description: ' + error.toString());
-}
+  remotePeerConnection.addEventListener('icecandidate', connectionHandler);
+  remotePeerConnection.addEventListener(
+    'iceconnectionstatechange', connectionChanged);
+  remotePeerConnection.addEventListener('addstream', gotRemoteMediaStream);
 
-function gotRemoteStream(e) {
-  // Add remoteStream to global scope so it's accessible from the browser console
-  window.remoteStream = remoteVideo.srcObject = e.stream;
-  trace('pc2 received remote stream');
-}
+  // Add local stream to connection and create offer to connect.
+  localPeerConnection.addStream(localStream);
+  trace('Added local stream to localPeerConnection.');
 
-function onCreateAnswerSuccess(desc) {
-  trace('Answer from pc2:\n' + desc.sdp);
-  trace('pc2 setLocalDescription start');
-  pc2.setLocalDescription(desc).then(
-    function() {
-      onSetLocalSuccess(pc2);
-    },
-    onSetSessionDescriptionError
-  );
-  trace('pc1 setRemoteDescription start');
-  pc1.setRemoteDescription(desc).then(
-    function() {
-      onSetRemoteSuccess(pc1);
-    },
-    onSetSessionDescriptionError
-  );
-}
+  trace('localPeerConnection createOffer start.');
+  localPeerConnection.createOffer(offerOptions)
+    .then(offerCreated).catch(setSessionDescriptionError);
+};
 
-function onIceCandidate(pc, event) {
-  if (event.candidate) {
-    getOtherPc(pc).addIceCandidate(
-      new RTCIceCandidate(event.candidate)
-    ).then(
-      function() {
-        onAddIceCandidateSuccess(pc);
-      },
-      function(err) {
-        onAddIceCandidateError(pc, err);
-      }
-    );
-    trace(getName(pc) + ' ICE candidate: \n' + event.candidate.candidate);
-  }
-}
+// End up call by closing connections and resetting peers.
+const hangupAction = () => {
+  localPeerConnection.close();
+  remotePeerConnection.close();
+  localPeerConnection = null;
+  remotePeerConnection = null;
+  hangupButton.disabled = true;
+  callButton.disabled = false;
+  trace('Ending call.');
+};
 
-function onAddIceCandidateSuccess(pc) {
-  trace(getName(pc) + ' addIceCandidate success');
-}
+// Add functionality to buttons.
```

Comment:
Maybe 'Add click event handlers for buttons.' or similar.

Reply:
Done.